ludwig/hyperopt/utils.py

```diff
@@ -207,7 +207,13 @@ def update_hyperopt_params_with_defaults(hyperopt_params: HyperoptConfigDict) ->
     set_default_value(hyperopt_params, METRIC, LOSS)
     set_default_value(hyperopt_params, GOAL, MINIMIZE)
 
-    set_default_values(hyperopt_params[EXECUTOR], {TYPE: RAY, NUM_SAMPLES: 1, MAX_CONCURRENT_TRIALS: AUTO})
+    set_default_values(
+        hyperopt_params[EXECUTOR],
+        {TYPE: RAY, NUM_SAMPLES: 1, MAX_CONCURRENT_TRIALS: AUTO},
+    )
+
+    if hyperopt_params[EXECUTOR].get("trial_function_resources") is None:
```

Comment:
Is this the same as `set_default_values(hyperopt_params[EXECUTOR]["trial_function_resources"], {"CPU": 1, "GPU": 0})`?

Reply:
I believe this is different because we want to wipe out all other keys with our custom dictionary. I think when I tried the above, it just added the `CPU` and `GPU` keys alongside the `hyperopt_resources` key.